pandas/core/generic.py

```diff
@@ -4251,12 +4251,24 @@ def _is_view(self) -> bool:
         """Return boolean indicating if self is view of another array"""
         return self._mgr.is_view
 
+    @staticmethod
+    def _check_copy_deprecation(copy):
+        if copy is not lib.no_default:
+            warnings.warn(
+                "The copy keyword is deprecated and will be removed in a future "
+                "version. Copy-on-Write is active in pandas since 3.0 which utilizes "
+                "a lazy copy mechanism that defers copies until necessary. Use "
+                ".copy() to make an eager copy if necessary.",
+                DeprecationWarning,
```

Comment:
`DeprecationWarning` because we're going to start following `DeprecationWarning` -> `FutureWarning`?

Reply:
Yep, my intention was to switch to FutureWarning in 3.2, but I suspect that copy=False is used by a lot of downstream libraries